pilot/pkg/serviceregistry/kube/controller/ambientindex.go

```diff
@@ -955,6 +956,24 @@ func (c *Controller) AdditionalPodSubscriptions(
 	return shouldSubscribe
 }
 
+// syncAllWorkloadsForAmbient refreshes all ambient workloads.
+func (c *Controller) syncAllWorkloadsForAmbient() {
+	if c.ambientIndex != nil {
+		namespaces := c.namespaces.List(metav1.NamespaceAll, klabels.Everything())
+		discoveryNamespaces := make([]string, 0, len(namespaces))
+		for _, ns := range namespaces {
+			if c.opts.DiscoveryNamespacesFilter != nil &&
+				c.opts.DiscoveryNamespacesFilter.Filter(ns.GetName()) {
+				discoveryNamespaces = append(discoveryNamespaces, ns.GetName())
+			}
```

Comment:
can we use DiscoveryNamespacesFilter.GetMembers

Reply:
Yes we do, I'll update and test it.

Reply:
cool, does this mean ambient works with discovery selector as well?

Reply:
I haven't looked at all the other places that may be influenced

Reply:
>Yes we do, I'll update and test it.

@hanxiaop miss push?

Reply:
> @howardjohn I have observed at least once a case where a new Istiod processes the Waypint/Gateway resource first, then the namespace. This results in the ambientindex storing /waypointIP when it should be network/waypointIP.
Even if we're not supporting multi-network this could break local traffic

@hzxuzhonghu I'm looking into waypoint network before fixing this PR, otherwise this PR will modify the waypoint network with an incorrect value.

Reply:
ic